homeassistant/components/roon/media_player.py

```diff
@@ -190,21 +190,32 @@ def _parse_volume(cls, player_data):
             "level": 0,
             "step": 0,
             "muted": False,
+            "fixed": True,
+            "incremental": False,
         }
 
         try:
             volume_data = player_data["volume"]
-            volume_muted = volume_data["is_muted"]
-            volume_step = convert(volume_data["step"], int, 0)
-            volume_max = volume_data["max"]
-            volume_min = volume_data["min"]
-            raw_level = convert(volume_data["value"], float, 0)
 
-            volume_range = volume_max - volume_min
-            volume_percentage_factor = volume_range / 100
+            volume_fixed = False
+            volume_incremental = volume_data["type"] == "incremental"
+            volume_muted = volume_data.get("is_muted", False)
 
-            level = (raw_level - volume_min) / volume_percentage_factor
-            volume_level = convert(level, int, 0) / 100
+            try:
```

Comment:
Two nested try-blocks, both catching KeyError is a bit hard to read IMHO.
Could you try to rewrite it? Maybe add helpers `fixed_volume` +`incremental_volume` which return True if certain keys are missing?

Reply:
I tried to leave the previous code (which was quite carefully crafted) - but agree that this now isn't at all easy to understand. 

I'll take another look!